packages/host/app/commands/search-cards.ts

```diff
@@ -0,0 +1,93 @@
+import { service } from '@ember/service';
+
+import flatMap from 'lodash/flatMap';
+
+import { Filter, assertQuery } from '@cardstack/runtime-common';
+
+import type * as BaseCommandModule from 'https://cardstack.com/base/command';
+
+import HostBaseCommand from '../lib/host-base-command';
+
+import type CardService from '../services/card-service';
+import type RealmServerService from '../services/realm-server';
+
+export class SearchCardsByTypeAndTitleCommand extends HostBaseCommand<
+  BaseCommandModule.SearchCardsByTypeAndTitleInput,
+  BaseCommandModule.SearchCardsResult
+> {
+  description = 'Search for card instances by type and/or title';
+
+  async getInputType() {
+    let commandModule = await this.loadCommandModule();
+    const { SearchCardsByTypeAndTitleInput } = commandModule;
+    return SearchCardsByTypeAndTitleInput;
+  }
+
+  protected async run(
+    input: BaseCommandModule.SearchCardsByTypeAndTitleInput,
+  ): Promise<BaseCommandModule.SearchCardsResult> {
+    if (!input.title && !input.cardType && !input.type) {
+      throw new Error(
+        'At least one of title, cardType, or type must be provided',
+      );
+    }
+    let filter = {} as any;
+    if (input.title) {
+      filter.contains = { title: input.title };
+    }
+    if (input.cardType) {
+      filter.eq = { _cardType: input.cardType };
+    }
+    if (input.type) {
+      filter.type = input.type;
+    }
+    return new SearchCardsByQueryCommand(this.commandContext).execute({
+      query: {
+        filter: filter as Filter,
```

Comment:
that's interesting that you don't need to suppply an `on` field which tells the search which card def you are talking about. something like this https://github.com/cardstack/boxel/blob/main/packages/runtime-common/tests/index-query-engine-test.ts#L156-L165. i would have thought we needed to provide a CardDef code ref in the `on` field.

Reply:
i think your use of `type` is why we don't have to say `on`--but i would think `type` would need to default to the base `CardDef` if no `type` is supplied

Reply:
The backend falls back to CardDef if not specified: https://github.com/cardstack/boxel/blob/e03162c5c2a4194ca25672153369fa723d3f2031/packages/runtime-common/index-query-engine.ts#L357